test/sqllogictest/regex.slt

```diff
@@ -547,3 +576,109 @@ query T
 SELECT regexp_replace('foobarbaz', 'no match', 'X')
 ----
 foobarbaz
+
+statement ok
+CREATE TABLE r(s string, regex string, replacement string);
+
+statement ok
+INSERT INTO r VALUES ('foobarbaz', 'b..', 'X'), ('foobarbaz', 'b(..)', '>${1}<'), ('FOOBARBAZ', 'b..', 'X'), ('FOOZARZAZ', 'b..', 'X');
+
+query T multiline
+EXPLAIN
+SELECT regexp_replace(s, 'b..', 'X') FROM r;
+----
+Explained Query:
+  Project (#3)
+    Map (regexp_replace["b..", case_insensitive=false, limit=1](#0, "X"))
+      ReadStorage materialize.public.r
+
+Target cluster: quickstart
+
+EOF
+
+query TT
+SELECT s, regexp_replace(s, 'b..', 'X') FROM r;
+----
+foobarbaz
+fooXbaz
+foobarbaz
+fooXbaz
+FOOBARBAZ
+FOOBARBAZ
+FOOZARZAZ
+FOOZARZAZ
+
+query T multiline
+EXPLAIN
+SELECT regexp_replace(s, 'b..', 'X', 'i') FROM r;
+----
+Explained Query:
+  Project (#3)
+    Map (regexp_replace["b..", case_insensitive=true, limit=1](#0, "X"))
+      ReadStorage materialize.public.r
+
+Target cluster: quickstart
+
+EOF
+
+query TT
+SELECT s, regexp_replace(s, 'b..', 'X', 'i') FROM r;
+----
+FOOBARBAZ
+FOOXBAZ
+foobarbaz
+fooXbaz
+foobarbaz
+fooXbaz
+FOOZARZAZ
+FOOZARZAZ
+
+query T multiline
+EXPLAIN
+SELECT regexp_replace(s, 'b..', 'X', 'ig') FROM r;
+----
+Explained Query:
+  Project (#3)
+    Map (regexp_replace["b..", case_insensitive=true, limit=0](#0, "X"))
+      ReadStorage materialize.public.r
+
+Target cluster: quickstart
+
+EOF
+
+query TT
+SELECT s, regexp_replace(s, 'b..', 'X', 'ig') FROM r;
+----
+FOOBARBAZ
+FOOXX
+foobarbaz
+fooXX
+foobarbaz
+fooXX
+FOOZARZAZ
+FOOZARZAZ
+
+query T multiline
+EXPLAIN
+SELECT regexp_replace(s, regex, replacement, 'ig') FROM r;
+----
+Explained Query:
+  Project (#3)
+    Map (regexp_replace(#0, #1, #2, "ig"))
+      ReadStorage materialize.public.r
+
+Target cluster: quickstart
+
+EOF
+
+query TT
+SELECT s, regexp_replace(s, regex, replacement, 'ig') FROM r;
+----
+FOOBARBAZ
+FOOXX
+foobarbaz
+fooXX
+FOOZARZAZ
+FOOZARZAZ
+foobarbaz
+foo>ar<>az<
```

Comment:
Would be great to add some test cases that show the error path in action. (with constant input, non-constant input, with `query error` and EXPLAIN)

You can make slt expect an error instead of some output rows by writing
```
query error xxxxxxxxxx
```
at the top of the test case (instead of something like `query TT`).

(And link https://github.com/MaterializeInc/materialize/issues/27355 in a comment, because the error msg will look weird in the EXPLAIN.)

Reply:
For the `EXPLAIN` tests, I'm getting:
`FAIL: error: running file regexr.slt: Couldn't split "EXPLAIN\nSELECT regexp_replace('foobarbaz', 'b(..', 'X', 'ig');\n---\nExplained Query (fast path):\n  Error \"invalid regular expression: regex parse error:\n    b(..\n     ^\nerror: unclosed group\n\nTarget cluster: quickstart\n\nEOF\n\nquery error invalid regular expression flag: x\nSELECT regexp_replace('foobarbaz', 'b(..', 'X', 'igx');\n\nstatement ok\nCREATE TABLE e(s string, regex string, replacement string);\n\nstatement ok\nINSERT INTO e VALUES ('foobarbaz', 'b(..', 'X');\n\nquery error invalid regular expression: regex parse error:\nSELECT s, regexp_replace(s, regex, replacement, 'ig') FROM e;\n\nquery error invalid regular expression flag: x\nSELECT regexp_replace(s, regex, replacement, 'igx') FROM e;\n" at \r?\n----`

`--rewrite-results` similarly fails. How can I write the expected output?

Reply:
@ggevay ^ in case you missed the comment

Reply:
Hmm, I don't see the problem from this error msg either. Could you please point me to a branch where you have this? I'd like to debug it.

Reply:
Here you go: https://github.com/sdht0/materialize/tree/sql-regexp-replace

Reply:
This branch seems to be the current state of the PR, where `regex.slt` is passing. I meant a branch where the above error is happening.

Reply:
Right too early in the morning :sweat_smile:
https://github.com/sdht0/materialize/tree/sql-regexp-replace-2

Reply:
New file: `test/sqllogictest/regexr.slt`

Reply:
Thanks! :)

Reply:
One `-` missing on line 16.

Reply:
Obvious in hindsight. Sorry, I was looking in the wrong place.

I've pushed the error path testing code all tests pass: `PASS: warning=1 success=73 total=74`. This warning is present even in `main` fyi.

Reply:
Ok, thanks!